Mathlib/AlgebraicGeometry/AlgebraicCycle.lean

```diff
@@ -0,0 +1,281 @@
+/-
+Copyright (c) 2025 Raphael Douglas Giles. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Raphael Douglas Giles
+-/
+import Mathlib.AlgebraicGeometry.Fiber
+import Mathlib.AlgebraicGeometry.Morphisms.Proper
+import Mathlib.Combinatorics.Quiver.ReflQuiver
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+import Mathlib.Topology.LocallyFinsupp
+
+/-!
+# Algebraic Cycles
+
+We define algebraic cycles on a scheme `X` to be functions `c : X → ℤ` whose support is
+locally finite.
+-/
+
+open AlgebraicGeometry Set Order LocallyRingedSpace Topology TopologicalSpace
+  CategoryTheory
+
+universe u v
+variable (R : Type*)
+         [CommRing R]
+         (i : ℕ)
+         {X Y : Scheme.{u}}
+
+/--
+Algebraic cycle on a scheme X.
+
+Note I am not certain that this should be an abbrev. I'm also not sure if these definitions
+should instead directly be about Function.locallyFinsuppWithin
+-/
+abbrev AlgebraicCycle (X : Scheme.{u}) := Function.locallyFinsuppWithin (⊤ : Set X) ℤ
```

Comment:
We probably want to have notions of cycles with arbitrary coefficients (which may not even have to be a group: the case of natural numbers is interesting).